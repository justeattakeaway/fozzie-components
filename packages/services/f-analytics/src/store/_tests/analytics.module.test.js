import AnalyticsModule from '../analyticsModule';
import {
    defaultState,
    modifieldState
} from '../../mixins/_tests/helpers/setup';

const { actions, mutations } = AnalyticsModule;
const { updatePlatformData } = actions;
let state = AnalyticsModule.state();

describe('AnalyticsModule', () => {
    it('should create default state when initialised.', () => {
        // Assert
        expect(state).toEqual(defaultState);
    });

    describe('mutations ::', () => {
        beforeEach(() => {
            // Arrange
            state = defaultState;
        });

        describe(`${updatePlatformData} ::`, () => {
            it('should update state with `platformData`', () => {
                // Act
                mutations.updatePlatformData(state, modifieldState);

                // Assert
                expect(state).toMatchSnapshot();
            });
        });
    });

    describe('actions ::', () => {
        describe(`${updatePlatformData} ::`, () => {
            it('should call the `updatePlatformData` mutation', () => {
                // Arrange
                const commit = jest.fn();

                // Act
                updatePlatformData({ commit }, modifieldState);

                // Assert
                expect(commit).toHaveBeenCalledWith('updatePlatformData', modifieldState);
            });
        });
    });
});
