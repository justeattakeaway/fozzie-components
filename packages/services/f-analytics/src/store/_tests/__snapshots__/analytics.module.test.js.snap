// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalyticsModule mutations :: ({
      commit
    }, platformData) => {
      commit('updatePlatformData', platformData);
    } :: should update state with \`platformData\` 1`] = `
Object {
  "platformData": Object {
    "platformData": Object {
      "appType": "test-appType",
      "applicationId": 9,
      "branding": "test-branding",
      "country": "zu",
      "currency": "zud",
      "environment": "test-environment",
      "instancePosition": "999",
      "jeUserPercentage": 88,
      "language": "ze",
      "name": "test-name",
      "userAgent": "test-userAgent",
      "version": "9.8.7.6",
    },
  },
}
`;
